pkg/cloud/gcp/gcs_kms.go

```diff
@@ -74,22 +77,46 @@ func MakeGCSKMS(ctx context.Context, uri string, env cloud.KMSEnv) (cloud.KMS, e
 
 	switch kmsURIParams.auth {
 	case "", cloud.AuthParamSpecified:
-		if kmsURIParams.credentials == "" {
-			return nil, errors.Errorf(
-				"%s is set to '%s', but %s is not set",
-				cloud.AuthParam,
-				cloud.AuthParamSpecified,
-				CredentialsParam,
-			)
+		if temporaryTokenEnabled.Get(&env.ClusterSettings().SV) {
+			if kmsURIParams.credentials == "" {
+				if kmsURIParams.temporaryToken == "" {
+					return nil, errors.Errorf(
+						"%s or %s must be set if %q is %q",
+						CredentialsParam,
+						TemporaryTokenParam,
+						cloud.AuthParam,
+						cloud.AuthParamSpecified,
+					)
+				}
+
+				// Create a static token source directly from TemporaryToken.
+				token := &oauth2.Token{AccessToken: kmsURIParams.temporaryToken}
+				credentialsOpt = append(credentialsOpt, option.WithTokenSource(oauth2.StaticTokenSource(token)))
+			} else {
+				authOption, err := createAuthOption(kmsURIParams.credentials)
```

Comment:
Is there a reason this handling of `credentials` is different from the one below in the else condition? I think with the if restructure this goes away anyways.